lib/HotModuleReplacementPlugin.js

```diff
@@ -145,7 +145,9 @@ module.exports = class HotModuleReplacementPlugin {
 							h: compilation.hash,
 							c: {}
 						};
-						for (let chunkId of Object.keys(records.chunkHashs)) {
+						/** @type {string|number} */
+						let chunkId;
+						for (chunkId of Object.keys(records.chunkHashs)) {
```

Comment:
When initializing a variable with `for..of` we can not override its type with `@type`. I had to move it out.

But the better solution is not to override a variable with a different type.

Reply:
That is the wrong type anyway. `Object.keys()` can _only_ produce `string`s, regardless of how much your key looks like a number.

Reply:
I understand this but in the next line `chunkId` is being casted to number using `+` operator and then rewritten to the same variable. TypeScript doesn't like that unless we make it clear that `chunkId` is type of `string|number`

Reply:
How about this?

https://github.com/webpack/webpack/pull/7020